src/ui/public/kfetch/kfetch.test.ts

```diff
@@ -104,8 +104,100 @@ describe('kfetch', () => {
       }).catch(e => {
         expect(e.message).toBe('Not Found');
         expect(e.res.status).toBe(404);
-        expect(e.res.url).toBe('myBase/my/path?a=b');
+        expect(e.res.url).toBe('http://localhost.com/myBase/my/path?a=b');
       });
     });
   });
+
+  describe('interceptors', () => {
+    afterEach(() => {
+      fetchMock.restore();
+      _resetInterceptors();
+    });
+
+    it('response: should modify response via interceptor', async () => {
+      fetchMock.get(matcherName, new Response(JSON.stringify({ foo: 'bar' })));
+      interceptors.push({
+        response: res => {
+          return {
+            ...res,
+            addedByInterceptor: true,
+          };
+        },
+      });
+
+      const resp = await kfetch({ pathname: 'my/path' });
+      expect(resp).toEqual({
+        addedByInterceptor: true,
+        foo: 'bar',
+      });
+    });
```

Comment:
Can we add a test verifying that a `Promise` is a valid return value?

```js
    it('response: should modify response via Promise interceptor', async () => {
      fetchMock.get(matcherName, new Response(JSON.stringify({ foo: 'bar' })));
      interceptors.push({
        response: res => {
          return new Promise((resolve) => {
            resolve({
              ...res,
              addedByInterceptor: true,
            });
          }
        },
      });

      const resp = await kfetch({ pathname: 'my/path' });
      expect(resp).toEqual({
        addedByInterceptor: true,
        foo: 'bar',
      });
    });
```

Reply:
This also begs the question, what's the expected behavior if an interceptor returns a rejected promise?